pkg/auth/providers/azure/clients/conversions.go

```diff
@@ -15,52 +15,56 @@ import (
 
 const Name = "azuread"
 
-// UserToPrincipal attempts to convert a raw user from Graph SDK to a Principal.
-// The user may be of type returned by Azure AD Graph API (deprecated and to be removed soon)
-// or Microsoft Graph API.
-func UserToPrincipal(user interface{}) (v3.Principal, error) {
-	var id, displayName, loginName string
+func (c azureADGraphClient) userToPrincipal(user interface{}) (v3.Principal, error) {
```

Comment:
Now that this method is on the client you should not have to use an interface{} as the type of the first arg. Would recommend changing this to type `graphrbac.User`.

similar comment for the rest of the functions in the file - we can use real types in the params now.